docs/kilnjs.md

```diff
@@ -0,0 +1,220 @@
+
+```
+  schema.enableSocialButtons.on('input', (val) => {
+    if(val) {
+      schema.shareServices.show();
+    } else {
+      schema.shareServices.hide();
+    }
+  });
+```
+
+### Validation
+
+By attaching events, one of the things you can use kilnjs for is to do field validation.
+
+```
+module.exports = (schema) => {
+  schema.title = new KilnInput(schema, 'title');
+
+  schema.title.on('keydown', (e) => {
+    // prevent numbers from being entered in the title field
+    if (!isNaN(e.key)) {
+      e.preventDefault();
+    }
+  });
+
+  return schema;
+};
+```
+
+---
+
+## Vuex Actions to Subscribe To
+
+The following are some of the Vuex actions that you can subscribe to using Kiln.js. The actions can be scoped so that they only trigger the subscription function when the action is triggered by a component of the same type as the subscribing component. In other words, if scope is set to true on a paragraph element, then the subscription function would only be called when a paragraph element triggers the Vuex action, but not when any other type of component triggers the Vuex action.  When the scope is set to false, the subscription function is called whenever _any_ component triggers the Vuex action.
+
+* ***OPEN_FORM*** - Triggered when a Kiln form is opened. Note that this can triggered by both a modal form _and_ an inline form, so just clicking into a paragraph or other inline field will trigger this action. The payload returned from this action contains a JSON representation of the Schema along with the component uri that was opened in the form.
+
+* ***CLOSE_FORM*** - Triggered when a Kiln form is closed. For inline fields this occurs when the field loses focus.  When the user clicks out of the field.  The payload returned from this action is undefined.
+
+* ***UPDATE_FORMDATA*** - Triggered when a user changes the data in a form field.  This happens "on change", _not_ "on save". The payload returns and object containing the path (field being changed) and data (the new value).
+
+* ***UPDATE_COMPONENT*** - Triggered when a component is saved.  The payload returns a json object containing all the fields in the component's schema, with the values for each of them.
+
+* ***UPDATE_PAGE_STATE*** - Triggered when the page state changes.  When it is published, unpublished, etc. The payload returned contains the page meta information, including published date/time, update date/time, history, users, etc.
+
+Often you might not wish to associate a subscription with a specific field, but rather to something more general.  You can instantiate a KilnInput without referencing a specific field. Of course, if it makes sense to create a connection between a field and the action involved with a subscription, you can use the subscribe function on a field.  It can be done either way.  It's just a matter of what makes the intent clearer.
```

Comment:
```suggestion
Often you might not wish to associate a subscription with a specific field, but rather to something more general.  You can instantiate a `KilnInput` without referencing a specific field. Of course, if it makes sense to create a connection between a field and the action involved with a subscription, you can use the subscribe function on a field. It can be done either way. It's just a matter of what makes the intent clearer.
```

Reply:
👍